pybop/optimisers/scipy_optimisers.py

```diff
@@ -1,85 +1,193 @@
+    options=dict(),
+)
 
-class SciPyMinimize(BaseOptimiser):
-    """
-    Adapts SciPy's minimize function for use as an optimization strategy.
 
-    This class provides an interface to various scalar minimization algorithms implemented in SciPy, allowing fine-tuning of the optimization process through method selection and option configuration.
+class BaseSciPyOptimiser(Optimisation):
+    """
+    A base class for defining optimisation methods from the SciPy library.
 
     Parameters
     ----------
-    method : str, optional
-        The type of solver to use. If not specified, defaults to 'Nelder-Mead'.
-        Options: 'Nelder-Mead', 'Powell', 'CG', 'BFGS', 'Newton-CG', 'L-BFGS-B', 'TNC', 'COBYLA', 'SLSQP', 'trust-constr', 'dogleg', 'trust-ncg', 'trust-exact', 'trust-krylov'.
-    bounds : sequence or ``Bounds``, optional
+    x0 : array_like
+        Initial position from which optimisation will start.
+    sigma0 : float, optional
+        Initial step size or standard deviation depending on the optimiser.
+    bounds : dict, sequence or scipy.optimize.Bounds, optional
         Bounds for variables as supported by the selected method.
-    maxiter : int, optional
-        Maximum number of iterations to perform.
+    **optimiser_kwargs : optional
+        Valid SciPy option keys and their values.
     """
 
-    def __init__(self, method=None, bounds=None, maxiter=None, tol=1e-5):
-        super().__init__()
-        self.method = method
-        self.bounds = bounds
-        self.tol = tol
-        self.options = {}
-        self._max_iterations = maxiter
+    def __init__(self, cost, **optimiser_kwargs):
+        super().__init__(cost, **optimiser_kwargs)
 
-        if self.method is None:
-            self.method = "Nelder-Mead"
-
-    def _runoptimise(self, cost_function, x0):
+    def _set_options(self, **optimiser_kwargs):
         """
-        Executes the optimization process using SciPy's minimize function.
+        Update the optimiser options and remove the corresponding entries from the
+        optimiser_kwargs dictionary in advance of passing to the parent class.
 
         Parameters
         ----------
-        cost_function : callable
-            The objective function to minimize.
+        **optimiser_kwargs : optional
+            Valid SciPy option keys and their values.
+
+        Returns
+        -------
+        optimiser_kwargs : dict
+            Remaining option keys and their values.
+        """
+        # Unpack nested values from SciPy options dictionary
+        if "options" in optimiser_kwargs.keys():
+            options_list = list(optimiser_kwargs["options"].keys())
+            for key in options_list:
+                if key not in optimiser_kwargs.keys():
+                    optimiser_kwargs[key] = optimiser_kwargs["options"].pop(key)
+                else:
+                    optimiser_kwargs["options"].pop(key)  # remove entry
+
+        # Keep max_iterations in preference to maxiter
```

Comment:
To confirm, we are taking preference of kwarg "max_iterations" over "maxiter"? If so this might be clearer:
```suggestion
        # Overwrite maxiter variable with kwarg max_iterations value
```